pkg/lib/authn/authenticator/password/provider.go

```diff
@@ -84,7 +85,7 @@ func (p *Provider) New(id string, userID string, passwordSpec *authenticator.Pas
 	}
 }
 
-// WithSpec return new authenticator pointer if password is changed
+// WithSpec return new authenticator pointer if password or expireAfter is changed
 // Otherwise original authenticator will be returned
 func (p *Provider) WithSpec(a *authenticator.Password, spec *authenticator.PasswordSpec) (*authenticator.Password, error) {
```

Comment:
Also rename this to UpdatePassword and take UpdatePasswordOptions. You may need to duplicate UpdatePasswordOptions in two packages, since you cannot import service.UpdatePasswordOptions in this package.